framework/src/util/AnyMap.cpp

```diff
@@ -1027,7 +1027,48 @@ bool any_map::operator==(const any_map& rhs) const
       case map_type::UNORDERED_MAP:
         return (*map.uo == *rhs.map.uo);
       case map_type::UNORDERED_MAP_CASEINSENSITIVE_KEYS:
-        return (*map.uoci == *rhs.map.uoci);
+      {
+        // In the case of our case insensitive keys map, there's an "oversight" in the
+        // std::unordered_map implementation such that if the 4th template argument is passed in
+        // (that is, the key-equality argument), it is only used when doing lookups in the map, not
+        // when std::unordered_map::operator== is used to check for equality between two
+        // maps. Inside the operator==, keys are compared with their own operator== rather than
+        // using the key equality template argument. In our case, this means that if I have a "key"
+        // in map1, and a "KEY" in map2, the maps will compare as inequal even though we have
+        // specified a case-insensitive comparator. So, we have to roll our own to make sure we get
+        // case-insensitive key comparisons.
```

Comment:
I'm not sure if this is an "oversight". Map equality, and specifically std::unordered_map equality, between two std::unordered_maps is solely dependent on whether or not the key-value pairs in one map are identical to another (albeit allowing for various permutations since the map is unordered).

The comparator we specify is used when finding keys and their associated values. Even in regular maps, it is my understanding that the comparator isn't used when checking equality; ordered maps similarly checks for equality of key-value pairs.